packages/app/src/components/Sidebar/Bookmarks.tsx

```diff
@@ -1,16 +1,118 @@
+const ACTIVE_PAGE = 1;
+
+type Props = {
+  pages: IPageHasId[]
+}
+
+const BookmarksItem = (props: Props) => {
+  const { pages } = props;
+
+  const generateBookmarkedPageList = pages.map((page) => {
+    const dPagePath = new DevidedPagePath(page.path, false, true);
+    const { latter: pageTitle, former: formerPagePath } = dPagePath;
+    return (
+      <div key={page._id}>
+        <li className="list-group-item list-group-item-action border-0 py-0 pr-3 d-flex align-items-center" id={`bookmark-item-${page._id}`}>
+          <a href={`/${page._id}`} className="grw-bookmarks-title-anchor flex-grow-1">
+            <p className={`text-truncate m-auto ${page.isEmpty && 'grw-sidebar-text-muted'}`}>{pageTitle}</p>
+          </a>
+        </li>
+        <UncontrolledTooltip
+          modifiers={{ preventOverflow: { boundariesElement: 'window' } }}
+          autohide={false}
+          placement="right"
+          target={`bookmark-item-${page._id}`}
+        >
+          { formerPagePath || '/' }
+        </UncontrolledTooltip>
+      </div>
+    );
+  });
+
+
+  return (
+    <>
+      <ul className="grw-bookmarks-list list-group p-3">
+        <div className="grw-bookmarks-item-container">
+          {generateBookmarkedPageList}
+        </div>
+      </ul>
+    </>
+  );
+};
+
+
+const Bookmarks = () : JSX.Element => {
+  const { t } = useTranslation();
+  const { data: currentUser } = useCurrentUser();
+  const { data: isGuestUser } = useIsGuestUser();
+  const [pages, setPages] = useState<IPageHasId[]>([]);
+
+  const getMyBookmarkList = useCallback(async() => {
+    // TODO: Remove pagination and apply  scrolling (not infinity)
+    const page = ACTIVE_PAGE;
+
+    try {
+      const res = await apiv3Get(`/bookmarks/${currentUser?._id}`, { page });
+      const { paginationResult } = res.data;
+      setPages(paginationResult.docs.map((page) => {
+        return {
+          ...page.page,
+        };
+      }));
+    }
+    catch (error) {
+      logger.error('failed to fetch data', error);
+      toastError(error, 'Error occurred in bookmark page list');
+    }
+  }, [currentUser]);
 
-const Bookmarks: FC = () => {
-  const { t } = useTranslation('');
+  useEffect(() => {
+    getMyBookmarkList();
+  }, [getMyBookmarkList]);
 
-  // TODO Add bookmarks item
+  const renderBar = () => {
+    if (isGuestUser) {
+      return (
+        <h3 className="pl-3">
+          { t('Not available for guest') }
+        </h3>
+      );
+    }
+    if (pages.length === 0) {
+      return (
+        <h3 className="pl-3">
+          { t('No bookmarks yet') }
+        </h3>
+      );
+    }
+    return null;
+  };
 
   return (
-    <div className="grw-sidebar-content-header p-3">
-      <h3 className="mb-0">{t('Bookmarks')}</h3>
-    </div>
+    <>
+      <div className="grw-sidebar-content-header p-3">
+        <h3 className="mb-0">{t('Bookmarks')}</h3>
+      </div>
+      { renderBar() }
+      <BookmarksItem pages={pages} />
```

Comment:
You seems like misunderstanding about my feedback.
I supposed to  the following code


In addition,  `Bar` is a just example and it's not should be used.
You need to change to appropriate name.
See:  https://www.techopedia.com/definition/16355/bar

```suggestion
        { isGuestUser
        ? (
          <h3 className="pl-3">
            { t('Not available for guest') }
          </h3>
        )
        : ( 
        // this method name is just an example
        { renderBar() }
        )  
      }
```